domain/secret/service/service.go

```diff
@@ -130,17 +133,28 @@ func (s *SecretService) CreateSecret(ctx context.Context, uri *secrets.URI, para
 		err := s.st.CreateUserSecret(ctx, params.Version, uri, p)
 		return errors.Annotatef(err, "creating user secret %q", uri.ID)
 	}
-	// TODO(secrets) - charm secrets
-	//var nextRotateTime *time.Time
-	//if params.RotatePolicy.WillRotate() {
-	//	nextRotateTime = params.RotatePolicy.NextRotateTime(s.clock.Now())
-	//}
-	return errors.NotImplemented
+
+	rotatePolicy := domainsecret.MarshallRotatePolicy(params.RotatePolicy)
+	p.RotatePolicy = &rotatePolicy
+	if params.RotatePolicy.WillRotate() {
```

Comment:
The provided `NextRotateTime` should not be overwritten.
```
p.NextRotateTime = params.NextRotateTime
if params.RotatePolicy.WillRotate() && p.NextRotateTime == nil {}

Reply:
There is no NextRotateTime that is provided in params. Juju always calculates it.